### PR TITLE
add libarchive 3.7.4.bcr.1 with tests and windows support

### PR DESCRIPTION
**Pull request by zaucy**
Based on https://github.com/zaucy/libarchive/tree/bazel-v3.7.4

* adds tests
* adds Windows support

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @dzbarsky, modules you maintain (libarchive) have been updated in this PR. Please review the changes.

**Comment by zaucy** [comment]
@dzbarsky I would love your opinion on these changes since you were the original maintainer. If this works out I intend to subumit these changes for 3.7.2 as well.

**Comment by DavidZbarsky-at** [comment]
Thanks for submitting! I'm about to go on vacation for a week, I will do my best to find time to look before that

**Comment by zaucy** [comment]
The presubmit is mostly green after I've excluded some tests (with added notes for any future readers.) 

However, there is a patch error only on ubuntu arm64 for some reason. If anyone has any insight on that, it would be very helpful.

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [inline modules/libarchive/3.7.3/patches/build_with_bazel.patch:183]: nit: I would usually inline the `SRCS` here. I find it works a lot better with buildozer/etc

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [inline modules/libarchive/3.7.3/patches/build_with_bazel.patch:1352]: having all the tests is awesome, thanks for doing this!

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [inline modules/libarchive/3.7.3/patches/build_with_bazel.patch:1552]: why make this one separate vs having a single `config.h` with some ifdefs for windows? Then you wouldn't need to vary the config filename either?

not a blocker, but it would be nice to resolve this so we don't need additional changes to the config.h includes

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [inline modules/libarchive/3.7.3/patches/build_with_bazel.patch:1712]: hmmm, this was working on macos before. What's not working now?
I think this is a blocker for landing

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (zaucy) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1552]: I just generated it with cmake and figured it had the most coverage. I could do what you're saying and add a bunch of preprocessor ifs, but it seemed very different than the 'generic' one. I could go either way 🤷

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (zaucy) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1712]: I re-enabled it on my fork to remind myself: https://github.com/zaucy/libarchive/actions/runs/8953745198/job/24592569883

```
tar/bsdtar.c:1078:50: error: expected ';' after top level declarator
        "  -f <filename>  Location of archive (default " _PATH_DEFTAPE ")\n"
```

I don't have a mac so I wasn't sure what was wrong with it. Works on Windows and Ubuntu on my machine and github actions.

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1552]: another option is to have a `copy_file` rule that creates a `config.h` based on the right platform. I'm leaning toward doing either that or handwriting the config.h, but both of those seem preferable to patching the libarchive source to support a dynamic config filename (especially if/when https://github.com/bazelbuild/bazel-central-registry/issues/1566 lands and we can do overlays for BUILD files instead of having to do patches)

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1712]: ah, that's right. I remember looking at https://github.com/libarchive/libarchive/blob/6818dd167cb190c76e0cfdeb40cd73aa22b96bd8/tar/bsdtar.c#L72-L83 and being a little confused how it's supposed to work on a mac.

It seems like not having it defined is fine, it just [requires patching the help message](https://github.com/bazelbuild/bazel-central-registry/blob/fcb922a1fa835db5bc1199d591c54b02dc73e2b3/modules/libarchive/3.7.2.bcr.2/patches/build_with_bazel.patch#L332-L333)

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (zaucy) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1552]: I think doing `copy_file` is a little overkill since these are _new_ files anyways. From reading #1566 it seems that having new non-bazel files would be supported.

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (zaucy) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1712]: ah! thanks. That is a little weird that it isn't an issue for them upstream 🤔

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (zaucy) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch]: I didn't realise it made working with buildozer worse. Thanks! It's updated now.

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1547]: Sorry, I think I could have been clearer. I agree that new files are supported, I'm just trying to avoid the need to patch the header name into the libarchive sources, which was why I suggested using the upstream name (`config.h`). `copy_file` is just one way to accomplish that, another one would be a `write_file` or `genrule` with a platform-dependent content

(BTW this comment is not blocking, just some thoughts to make future maintenance easier)

File: modules/libarchive/3.7.3/MODULE.bazel
Comment (dzbarsky) [inline modules/libarchive/3.7.3/MODULE.bazel:3]: would you mind doing this on 3.7.4? That's the latest version in the BCR now

File: modules/libarchive/3.7.3/presubmit.yml
Comment (dzbarsky) [inline modules/libarchive/3.7.3/presubmit.yml:8]: how did you generate the patch? I've seen patches generated with `git diff` work but `git show` not work in the past. Maybe it would help to trim the header lines?

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [inline modules/libarchive/3.7.3/patches/build_with_bazel.patch:1728]: yeah I have no clue how this works upstream; maybe apple patches this as well :)

**Review by dzbarsky — APPROVED** [review]
I think this can go in as-is if you'd like, but I'd prefer to generate the config as `config.h` and reduce the patching, and a 3.7.4 base would be great as well!

Thanks for helping on this one, much appreciated

**Review by bazel-io — APPROVED** [review]
Hello @bazelbuild/bcr-maintainers, all modules in this PR have been approved by their maintainers. Please take a final look to merge this PR.

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (zaucy) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1547]: oh we don't have to patch the header name since libarchive supports a custom path with `PLATFORM_CONFIG_H` define!

File: modules/libarchive/3.7.3/presubmit.yml
Comment (zaucy) [reply modules/libarchive/3.7.3/presubmit.yml:8]: I just used `git diff`. I have no idea why its not working. This is how I'm currently generating it https://github.com/zaucy/libarchive/blob/7cb94b082c11bd171ac219fbb89f9517f7f23526/bazel/update_bcr.nu#L25

File: modules/libarchive/3.7.3/MODULE.bazel
Comment (zaucy) [reply modules/libarchive/3.7.3/MODULE.bazel:3]: I was thinking of doing `3.7.4` as a follow up once we felt this patch was good, but how about I just include both `3.7.3` and `3.7.4.bcr.1` in this PR? :)

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (dzbarsky) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1547]: oh I see, were you previously patching it? Not sure why I thought you were. I think it's probably still slightly less code overall to not have the platform-dependent names, but not a huge deal either way

File: modules/libarchive/3.7.3/MODULE.bazel
Comment (dzbarsky) [reply modules/libarchive/3.7.3/MODULE.bazel:3]: whatever's easier for you :)

File: modules/libarchive/3.7.3/presubmit.yml
Comment (dzbarsky) [reply modules/libarchive/3.7.3/presubmit.yml:8]: hmm, maybe @Wyverald has an idea?

File: modules/libarchive/3.7.3/presubmit.yml
Comment (thesayyn) [inline modules/libarchive/3.7.3/presubmit.yml:16]: hmm, why does libarchive needs tls?

**Review by bazel-io — CHANGES_REQUESTED** [review]
Require module maintainers' approval.

File: modules/libarchive/3.7.3/presubmit.yml
Comment (thesayyn) [reply modules/libarchive/3.7.3/presubmit.yml:16]: oh for, crypto library...

Comment (zaucy) [reply modules/libarchive/3.7.3/presubmit.yml:16]: Yes. the config is not new, only diff here is that I'm actually getting the presubmit to run with it.

Comment (thesayyn) [reply modules/libarchive/3.7.3/presubmit.yml:16]: makes sense

File: modules/libarchive/3.7.3/MODULE.bazel
Comment (zaucy) [reply modules/libarchive/3.7.3/MODULE.bazel:3]: once all the other conversations are resolved I'll add `3.7.4.bcr.1` :+1:

File: modules/libarchive/3.7.3/patches/build_with_bazel.patch
Comment (zaucy) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1547]: I did patch one spot since the test util file didn't use the `PLATFORM_CONFIG_H`, but I intend to upstream that change soon.

> I think it's probably still slightly less code overall to not have the platform-dependent names

I do agree having the 1 bazel config header would be nicer. I might give it a try, but the generated cmake one might be more accurate/maintainable.

Comment (dzbarsky) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1547]: Yeah I think using the cmake header is probably fine, I was thinking something like this
```
copy_file(
  name = "gen_config",
  outs = "config.h",
  src = select({
    "@platforms//os:windows": "cmake_generated_windows_config.h",
    "//conditions:default": "cmake_generated_windows_config.h",
     }),
 )
 ```

Comment (zaucy) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1547]: ahh I see what you're saying. Just to consolidate the files into one. I'm fine to add that change. Is there a specific reason you like having it as `config.h` instead of having the platform and 'bazel' in the name? 

Having the platform name almost makes it clearer to me which one I'm using, so I wasn't bothered by it.

I would like to avoid using `HAVE_CONFIG_H` since `libarchive` uses the config file in their public headers to avoid issues like this: https://github.com/bazelbuild/bazel-central-registry/pull/1917#issuecomment-2093984116, but I can still use `PLATFORM_CONFI_H` with it being `config.h` so that shouldn't be an issue.

Comment (dzbarsky) [reply modules/libarchive/3.7.3/patches/build_with_bazel.patch:1547]: Ah I see. I was going to suggest local_defines but doesn't sound like that works here. Maybe upstream would take a patch renaming it to HAVE_LIBARCHIVE_CONFIG_H? I'm surprised non-bazel builds don't suffer from this

Anyway, what you wrote in the last comment makes sense

**Comment by peakschris** [comment]
This fix is required to allow rules_js to work on windows. Can it be moved forward?

**Comment by alexeagle** [comment]
@dzbarsky we do still need this, right?

**Comment by zaucy** [comment]
yes currently https://github.com/bazelbuild/bazel-central-registry/pull/1916#discussion_r1594523428 is making me hesitant to say it's **done**. After that's resolved I'm happy to update this PR with `3.7.4.bcr.1` added as well.

**Comment by bazel-io** [comment]
Hello @dzbarsky, modules you maintain (libarchive) have been updated in this PR. Please review the changes.

**Comment by zaucy** [comment]
Unfortunately the patches aren't applying for arm64 in the presubmit and I'm not sure why. If anyone has any insight it would be much appreciated.

https://buildkite.com/bazel/bcr-presubmit/builds/6046#018feebc-e114-416d-ab7c-66064716e347
```log
Applying build_with_bazel.patch:
--
  | The next patch would create the file .bazelrc,
  | which already exists!  Applying it anyway.
  | patching file .bazelrc
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file .bazelrc.rej
  | The next patch would create the file BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file BUILD.bazel.rej
  | The next patch would create the file MODULE.bazel,
  | which already exists!  Applying it anyway.
  | patching file MODULE.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file MODULE.bazel.rej
  | The next patch would create the file WORKSPACE.bzlmod,
  | which already exists!  Applying it anyway.
  | patching file WORKSPACE.bzlmod
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file WORKSPACE.bzlmod.rej
  | The next patch would create the file bazel/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file bazel/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file bazel/BUILD.bazel.rej
  | The next patch would create the file bazel/config.bzl,
  | which already exists!  Applying it anyway.
  | patching file bazel/config.bzl
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file bazel/config.bzl.rej
  | The next patch would create the file bazel/libarchive_test.bzl,
  | which already exists!  Applying it anyway.
  | patching file bazel/libarchive_test.bzl
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file bazel/libarchive_test.bzl.rej
  | The next patch would create the file cat/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file cat/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file cat/BUILD.bazel.rej
  | The next patch would create the file cpio/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file cpio/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file cpio/BUILD.bazel.rej
  | The next patch would create the file libarchive/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file libarchive/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file libarchive/BUILD.bazel.rej
  | The next patch would create the file libarchive/test/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file libarchive/test/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file libarchive/test/BUILD.bazel.rej
  | The next patch would create the file libarchive_bazel_generic_config.h,
  | which already exists!  Applying it anyway.
  | patching file libarchive_bazel_generic_config.h
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file libarchive_bazel_generic_config.h.rej
  | The next patch would create the file libarchive_bazel_windows_config.h,
  | which already exists!  Applying it anyway.
  | patching file libarchive_bazel_windows_config.h
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file libarchive_bazel_windows_config.h.rej
  | The next patch would create the file libarchive_fe/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file libarchive_fe/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file libarchive_fe/BUILD.bazel.rej
  | The next patch would create the file tar/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file tar/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file tar/BUILD.bazel.rej
  | patching file tar/bsdtar.c
  | The next patch would create the file test_utils/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file test_utils/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file test_utils/BUILD.bazel.rej
  | patching file test_utils/test_common.h
  | The next patch would create the file unzip/BUILD.bazel,
  | which already exists!  Applying it anyway.
  | patching file unzip/BUILD.bazel
  | Hunk #1 FAILED at 1.
  | 1 out of 1 hunk FAILED -- saving rejects to file unzip/BUILD.bazel.rej
  | Traceback (most recent call last):
  | File "bcr_presubmit.py", line 513, in <module>
  | sys.exit(main())
  | File "bcr_presubmit.py", line 504, in main
  | repo_location, config_file = prepare_test_module_repo(args.module_name, args.module_version)
  | File "bcr_presubmit.py", line 245, in prepare_test_module_repo
  | apply_patch(source_root, source["patch_strip"], patch_file)
  | File "bcr_presubmit.py", line 202, in apply_patch
  | subprocess.run(
  | File "/usr/lib/python3.8/subprocess.py", line 516, in run
  | raise CalledProcessError(retcode, process.args,
  | subprocess.CalledProcessError: Command '['patch', '-f', '-p1', '-i', PosixPath('/workdir/modules/libarchive/3.7.4.bcr.1/patches/build_with_bazel.patch')]' returned non-zero exit status 1.
```

File: modules/libarchive/3.7.4.bcr.1/presubmit.yml
Comment (meteorcloudy) [inline modules/libarchive/3.7.4.bcr.1/presubmit.yml:8]: We don't have good support for this platform on Bazel CI, please remove it.

Comment (DavidZbarsky-at) [reply modules/libarchive/3.7.4.bcr.1/presubmit.yml]: @meteorcloudy can we get some sort of linux arm platform that is supported? A lot of us are deploying software to Graviton these days for cost efficiency reasons and it's unfortunate to not have it tested upstream; especially for C code it can have a lot of architecture-specific bits

**Comment by DavidZbarsky-at** [comment]
@meteorcloudy can we get some sort of linux arm platform that is supported? A lot of us are deploying software to Graviton these days for cost efficiency reasons and it's unfortunate to not have it tested upstream; especially for C code it can have a lot of architecture-specific bits

**Comment by meteorcloudy** [comment]
@fweikert What is the latest status of https://github.com/bazelbuild/continuous-integration/pull/1493? Is it possible to have Linux arm64 machines on our GCP project?

**Comment by fweikert** [comment]
Yeah, it's possible, I just didn't have time to fix all the weird breakages.

**Comment by peakschris** [comment]
@dzbarsky @meteorcloudy could this go in soon? Without it we are blocked in using rules_js on windows. Sorry for the nag and thanks in advance!

**Review by DavidZbarsky-at — APPROVED** [review]
Assuming it actually works on Linux arm, I think it can go in. It would be great to get CI coverage though

**Review by Wyverald — APPROVED** [review]

**Comment by alexeagle** [comment]
@DavidZbarsky-at the prebuilt release for this failed:

```
 In file included from external/libarchive~/libarchive/archive_hmac.c:32:
external/libarchive~/libarchive/archive_hmac_private.h:51:10: fatal error: 'CommonCrypto/CommonHMAC.h' file not found
   51 | #include <CommonCrypto/CommonHMAC.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```
https://github.com/aspect-build/bsdtar-prebuilt/actions/runs/9529065254

Is it non-hermetic and some crypto system library has to be installed on the runner?

**Comment by DavidZbarsky-at** [comment]
> @DavidZbarsky-at the prebuilt release for this failed:
> 
> ```
>  In file included from external/libarchive~/libarchive/archive_hmac.c:32:
> external/libarchive~/libarchive/archive_hmac_private.h:51:10: fatal error: 'CommonCrypto/CommonHMAC.h' file not found
>    51 | #include <CommonCrypto/CommonHMAC.h>
>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
> 1 error generated.
> ```
> 
> https://github.com/aspect-build/bsdtar-prebuilt/actions/runs/9529065254
> 
> Is it non-hermetic and some crypto system library has to be installed on the runner?

The CommonCrypto include was patched out in my original version and I guess the patch was not carried forward. I'm traveling for the next few weeks, @zaucy do you have a chance to fix it up?

**Comment by zaucy** [comment]
I'll take a look.

**Comment by zaucy** [comment]
Ah I see. I didn't notice the source patch. This is the one you're talking about right?

```diff
--- a/libarchive/archive_hmac_private.h
+++ b/libarchive/archive_hmac_private.h
@@ -40,12 +40,13 @@
  */
 int __libarchive_hmac_build_hack(void);
 
-#ifdef __APPLE__
-# include <AvailabilityMacros.h>
-# if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
-#  define ARCHIVE_HMAC_USE_Apple_CommonCrypto
-# endif
-#endif
+// Don't compile against CommonCrypto.
+// #ifdef __APPLE__
+// # include <AvailabilityMacros.h>
+// # if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
+// #  define ARCHIVE_HMAC_USE_Apple_CommonCrypto
+// # endif
+// #endif
```

I understand wanting to make bsdtar hermetic because its being used as a tool, but shouldn't we be avoiding modifying sources when submitting modules to the BCR? This feels like a patch that should be either done in a patch in the repo that wants it to be hermetic (e.g. https://github.com/aspect-build/bsdtar-prebuilt) and/or submitted as a feature request upstream to libarchive.

Regardless, I'll be fixing it since it was in the patch prior to this one.

**Comment by dzbarsky** [comment]
> Ah I see. I didn't notice the source patch. This is the one you're talking about right?
> 
> ```diff
> --- a/libarchive/archive_hmac_private.h
> +++ b/libarchive/archive_hmac_private.h
> @@ -40,12 +40,13 @@
>   */
>  int __libarchive_hmac_build_hack(void);
>  
> -#ifdef __APPLE__
> -# include <AvailabilityMacros.h>
> -# if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
> -#  define ARCHIVE_HMAC_USE_Apple_CommonCrypto
> -# endif
> -#endif
> +// Don't compile against CommonCrypto.
> +// #ifdef __APPLE__
> +// # include <AvailabilityMacros.h>
> +// # if MAC_OS_X_VERSION_MAX_ALLOWED >= 1060
> +// #  define ARCHIVE_HMAC_USE_Apple_CommonCrypto
> +// # endif
> +// #endif
> ```
> 
> I understand wanting to make bsdtar hermetic because its being used as a tool, but shouldn't we be avoiding modifying sources when submitting modules to the BCR? This feels like a patch that should be either done in a patch in the repo that wants it to be hermetic (e.g. https://github.com/aspect-build/bsdtar-prebuilt) and/or submitted as a feature request upstream to libarchive.
> 
> Regardless, I'll be fixing it since it was in the patch prior to this one.

Yep, that's the one. I don't have a super strong philosophical opinion on where the patch should live but compiling against and OSX SDK is a common pain point so I suspect others may want this patch as well.

I suppose if we're very motivated we could patch it to wrap this in another #define and allow configuring it with a build flag? Or add the feature upstream as you suggest

**Comment by zaucy** [comment]
Some kind of `#define` in the upstream that we could configure would probably be the best call imo!

anyways here's the fix: https://github.com/bazelbuild/bazel-central-registry/pull/2271

**Comment by alexeagle** [comment]
hey @zaucy - any idea why we get 

```
Action failed to execute: java.io.IOException: ERROR: src/main/native/windows/process.cc(202): CreateProcessW("C:\users\runneradmin\_bazel_runneradmin\snmkx6sz\execroot\_main\external\aspect_bazel_lib~~toolchains~bsd_tar_windows_amd64\tar.exe" --create --file bazel-out/x64_windows-fastbuild/bin/tar_empty.tar @bazel-out/x64_windows-fastbuild/bin/_tar_empty.mtree.txt): This version of %1 is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.
```
when running this binary on windows?
https://github.com/bazel-contrib/bazel-lib/actions/runs/10913564506/job/30290275800

**Comment by alexeagle** [comment]
Oh it's an ELF-format binary.

**Pull request merged**
